hypershift-operator/controllers/manifests/ignitionserver/manifests.go

```diff
@@ -0,0 +1,121 @@
+package ignitionserver
+
+import (
+	routev1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	k8sutilspointer "k8s.io/utils/pointer"
+)
+
+const (
+	resourceName = "ignition-server"
+)
+
+func Route(namespace string) *routev1.Route {
+	return &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      resourceName,
+		},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: resourceName,
+			},
+		},
+	}
+}
+
+func ReconcileRoute(route *routev1.Route) error {
+	route.Spec.To = routev1.RouteTargetReference{
+		Kind: "Service",
+		Name: resourceName,
+	}
+	return nil
+}
+
+func Service(namespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      resourceName,
+		},
+	}
+}
+
+func ReconcileServiceClusterIP(mcsService *corev1.Service) error {
+	mcsService.Spec.Ports = ServicePorts()
+	mcsService.Spec.Selector = ServiceSelector()
+	mcsService.Spec.Type = corev1.ServiceTypeClusterIP
+	return nil
+}
+
+func ServicePorts() []corev1.ServicePort {
```

Comment:
Not a manifest, should be inlined into the ClusterIPService reconcile function (which itself should be moved to the CreateOrUpdate call)